### ISSUE TITLE
interpreted text dropped from label titles stored in objects.inv

### ISSUE THREAD

**Issue by embray** [sphinx-doc/sphinx#5382]
### Problem
If the title of a section associated with a label contains interpreted text, the entire portion of the title that's interpreted is removed from the title.  This can be seen in the Sage docs (which is a collection of multiple Sphinx docs with intersphinx links between them).  There is a [page](http://doc.sagemath.org/html/en/reference/padics/sage/rings/padics/tutorial.html) whose raw rst markup contains:

```rst
.. _sage.rings.padics.tutorial:

Introduction to the `p`-adics
=============================

.. This file has been autogenerated.


.. automodule:: sage.rings.padics.tutorial
   :members:
   :undoc-members:
   :show-inheritance:
```

When referencing this label from another document, as in [this page](http://doc.sagemath.org/html/en/tutorial/tour_numtheory.html#adic-numbers) you can see that the part of the title in backticks is removed entirely.

When I manually loaded up the `objects.inv` for the first document, I see that:

```python
>>> invv = InventoryFile.load(open('local/share/doc/sage/html/en/reference/padics/objects.inv', 'rb'), 'local/share/doc/sage/html/en/reference/padics/', os.path.join)
>>> invv['std:label']['sage.rings.padics.tutorial']
(u'Sage Reference Manual: p-Adics', u'8.4.beta3', u'local/share/doc/sage/html/en/reference/padics/sage/rings/padics/tutorial.html#sage-rings-padics-tutorial', u'Introduction to the -adics')
```

although it's possible the problem starts someplace earlier.

#### Procedure to reproduce the problem

A procedure to reproduce the problem would be to create a page like the one above, and then produce its objects.inv.

#### Expected results

I'm not sure.  I suspect the current behavior may be intentional, though it is misleading.  Note that even if I explicitly put the role name in the title, like

```
:math:`p`-adics
```

it does not help.  I suspect the reason for this is that, being a title for an intersphinx link, a role from one document might not be interpreted correctly (if at all) as the same role in a different document.

I would rather have the option to either explicitly keep interpreted text along with their roles in intersphinx titles, *or* at least the option to keep the text inside the backticks without dropping it entirely.

### Environment info
- OS: Linux
- Python version: 2.7.15
- Sphinx version: 1.7.6

**Comment by embray** [comment]
(FWIW I'd be happy to supply a PR with a fix, but first would have to agree on what we're fixing.)

**Comment by tk0miya** [comment]
I think extracting text is better for such case.

**Comment by embray** [comment]
I think so too.  Arguably it's something one might want to make an option.  But unless someone asks for such an option simply extracting the plain text seems like the most sensible default.

**Comment by tk0miya** [comment]
Note: Now Sphinx uses node's name a.k.a `node['names']` as a link title. And it comes from raw source of section titles.
https://github.com/sphinx-doc/sphinx/blob/e8e0f93c3fc7a9f8be4a91c13c6ec49672347a1e/sphinx/domains/std.py#L609

**Comment by tk0miya** [comment]
I tried to fix this today. But I can't reproduce the situation. It seems the interpreted text is rendered as plain text in inventory file.

Indeed, rawtext of the title is used to the "name" field as I commented above. But "title" colum does not contain them.
```
(tmp) [146] tkomiya@deneb> python -m sphinx.ext.intersphinx _build/html/objects.inv
...
std:label
...
	indices and tables                       Indices and tables                      : index.html#indices-and-tables
	introduction to the :math:`p`-adics      Introduction to the p-adics             : index.html#sage-rings-padics-tutorial2
	introduction to the `p`-adics            Introduction to the p-adics             : index.html#introduction-to-the-p-adics
...
```

**Comment by embray** [comment]
Hmm, thanks for looking into it. Maybe the issue is already fixed somehow (I was using 1.7.x) or perhaps there's a funky interaction with an extension. I'll have to double check that and get back to you.

**Comment by tk0miya** [comment]
I could reproduced finally. It seems math role are ignored in the title.
I can reproduce the case with following settings with v1.7.x:
```
default_role = 'math'
```

But, fortunately, it has been already fixed in 1.8.x series. Since 1.8.0, we've integrated math extensions into sphinx-core. At that time, they are much refactored.

So I'm closing this issue now. Sorry for late investigation.
Please let me know if you got the same trouble with newer version of Sphinx.
Thanks,
